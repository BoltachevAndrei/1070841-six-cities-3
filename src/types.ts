export enum SortTypeList {
  POPULAR = `Popular`,
  LOW_TO_HIGH = `Price: low to high`,
  HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`
};

export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`
};

export type City = {
  location: {
    latitude: number,
    longitude: number,
    zoom: number
  },
  name: string
};

export type User = {
  id: number,
  name: string,
  avatar: string,
  isSuper: boolean,
  email?: string
};

export type Offer = {
  id: number,
  city: City,
  isPremium: boolean,
  images: Array<string>,
  price: number,
  isBookmarked: boolean,
  rating: number,
  title: string,
  description: string,
  features: {
    entire: string,
    bedrooms: string,
    adults: string
  },
  goodsInside: Array<string>,
  host: User,
  coordinates: Array<number>,
  zoom: number,
  previewImage: string
};

export type Comment = {
  text: string,
  date: Date,
  id: number,
  rating: number,
  user: User
};
