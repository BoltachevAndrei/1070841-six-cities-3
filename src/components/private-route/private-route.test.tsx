import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import history from '../../history';
import {AuthorizationStatus} from '../../types';
import PrivateRoute from './private-route';

const mockStore = configureStore([]);

const render = () => <div>Private route</div>;

it(`PrivateRoute with authorization renders Private route`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus: AuthorizationStatus.AUTH
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              render={render}
            />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`PrivateRoute with no authorization renders nothing`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus: AuthorizationStatus.NO_AUTH
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              render={render}
            />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
