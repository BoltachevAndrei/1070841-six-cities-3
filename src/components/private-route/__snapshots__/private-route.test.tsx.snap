// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute with authorization renders Private route 1`] = `
<div>
  Private route
</div>
`;

exports[`PrivateRoute with no authorization renders nothing 1`] = `null`;
